src/elf.rs

```diff
@@ -814,14 +760,19 @@ impl<'a> Elf<'a> {
 
         // alignment required for below
         let mut symtab_offset = strtab_offset + self.sizeof_strtab as u64;
-        let sizeof_symbol = Symbol::size(self.ctx.container) as u64;
-        Self::align(&mut symtab_offset, sizeof_symbol);
-        let mut reloc_offset = symtab_offset + sizeof_symtab as u64;
-        let sizeof_relocation = Relocation::size(true, self.ctx) as u64;
-        Self::align(&mut reloc_offset, sizeof_relocation);
+        let symtab_align = self.ctx.size() as u64;
+        Self::align(&mut symtab_offset, symtab_align);
+        let mut symtab_shndx_offset = symtab_offset + sizeof_symtab as u64;
+        let symtab_shndx_align = 4;
+        if sizeof_symtab_shndx > 0 {
+            Self::align(&mut symtab_shndx_offset, symtab_shndx_align);
+        }
+        let mut reloc_offset = symtab_shndx_offset + sizeof_symtab_shndx;
+        let reloc_align = self.ctx.size() as u64;
+        Self::align(&mut reloc_offset, reloc_align);
         let mut sh_offset = reloc_offset + sizeof_relocs as u64;
-        let sizeof_shdr = Section::size(&self.ctx) as u64;
-        Self::align(&mut sh_offset, sizeof_shdr);
+        let shdr_align = self.ctx.size() as u64;
```

Comment:
Similarly, what method is being called here? Why was this changed from `Section::size` ? Even if they are identical, it is less readable imho.